server/src/main/java/org/elasticsearch/transport/RemoteConnectionStrategy.java

```diff
@@ -105,10 +105,14 @@ public int getNumberOfChannels() {
             Setting.Property.NodeScope,
             Setting.Property.Dynamic));
 
+    // this setting is intentionally not registered, it is only used in tests
+    public static final Setting<Integer> REMOTE_MAX_CONNECTION_QUEUE_SIZE =
+        Setting.intSetting("cluster.remote.max_connection_queue_size", 100, Setting.Property.NodeScope);
```

Comment:
I don't think there was a lot of thought to the connection listener limit. If there is a strong reason to increase it past 100 we could probably do that. Also does does this name make sense? We only allow a single connection round at a time. Should the name be `cluster.remote.max_pending_connection_listeners`?

Reply:
> Should the name be cluster.remote.max_pending_connection_listeners?
> Should the name be cluster.remote.max_pending_connection_listeners?

++. I renamed it in https://github.com/elastic/elasticsearch/pull/55314/commits/f9c807f7a1b1acd387b8f3901941a8bc5ad5842e.

> I don't think there was a lot of thought to the connection listener limit. If there is a strong reason to increase it past 100 we could probably do that.

Yeah, I think we chose this value quite arbitrarily. I think it's fine to increase this value as we should not have many concurrent remote searches, and CCR will retry on this error anyway. I've increased this to 1000. WDYT?